### DISCUSSION TITLE
uninstall oh my zsh

### DISCUSSION THREAD

**Discussion by vmaryanne82** [ohmyzsh/ohmyzsh#11602] [Help] [answered]
I personally hate this tool. I tried everything possible under the google search to uninstall this. Nothing works.

I keep getting this command - zsh: command not found:  for everything. Even to open vi or touch ./zschrc to update plugins.

I have run uninstall_oh_my_zsh it says uninstall_oh_my_zsh:1: command not found: env

I have tried source ~/.oh-my-zsh/tools/uninstall.sh. It auto breaks at Line 13 It doesnt even give me the option to type Yes or No and auto cancels uninstallation process.

This is affecting all my other terminal in my M1 Mac and my work. I cant run any command on my terminal. Can you please help me uninstall this off my system.

**Comment by carlosala** [comment, marked as answer]
Removing the line `source oh-my-zsh.sh`from your `zshrc` should do the trick. You can also remove ~/.oh-my-zsh directory.

**Comment by RVasquez20** [comment]
You can remove ~/.oh-my-zsh directory.

**Comment by rwmitchell** [comment]
Wouldn't it be more important to restore the original .zshrc file that OMZ made a backup copy of before replacing?  And wouldn't it be that file which likely has the missing PATH settings which is now causing all the issues?

**Reply by RukshanJS** [reply]
When i `echo` ed the env vars that i were using before installing this tool, now it returns nothing. I would naturally expect this tool to just enhance the user experience, by adding some value to it, and not replace the current configs. I maybe wrong here but when installing the tool does it really "backup and forget"s the older env vars of the user? If that's the case, I might have to uninstall the tool sorry. Can someone please explain this? I'm fairly certain i'm doing something wrong here because that cannot be the default behavior right?

**Comment by prrtk** [comment]
Yes, You can remove ~/.oh-my-zsh directory.

**Reply by prrtk** [reply]
that'll fix the problem ig.

**Comment by ferdnyc** [comment]
I just submitted a PR (#11709) which may also have helped with this:  It uses the full `/usr/bin/env` path in `uninstall_oh_my_zsh()`, rather than a bare `env`, so that it should still run even if the `$PATH` is broken (as seemed to be the case here).

**Reply by ferdnyc** [reply]
Hrm. Actually that won't help, since it still won't find `sh`.

Surprisingly, though, having `/usr/bin/env` _unset_ the path (`/usr/bin/env -u PATH ZSH="$ZSH"`) seems to reset it to the default (`/usr/local/bin:/usr/bin`, in my case on Fedora 38). That then allows a bare `sh "$ZSH/tools/uninstall.sh"` to run. I wonder if I should add a `-u PATH` to the PR, as well?

**Comment by mdashikulislam** [comment]
You can run the following command:
`sh ~/.oh-my-zsh/tools/uninstall.sh`

You will be asked the following:
`Are you sure you want to remove Oh My Zsh? [y/N] `
Then enter `y` for "yes" and press `Enter`. This will restore `.zshrc` and remove the `~/.oh-my-zsh/` folder.

**Reply by scifithedev** [reply]
The best answer ever. Thank you

**Reply by mdashikulislam** [reply]
thank you

**Reply by FerGimenezRoglia** [reply]
esto si que funciona gracias

**Comment by JimBimBum** [comment]
i hated this tool so much, annoying intrusive "would you like to update?" worst part is it causes my terminals to crash after the recent updates to my pc. unacceptable and outright dangerous for me to have allowed installed if it wont even let me use ttw.

**Comment by BasamAlasaly** [comment]
I tried every solution here, including uninstalling, and nothing worked. For anyone who doesn't want to remove Oh My Zsh but wants to kill the compinit error for good, this is the only thing that fixed it for me.

### What Didn't Work

- Running `chmod` on any directory.
- Deleting the `.zcompdump` file.
- Adding `ZSH_DISABLE_COMPFIX=true` to my `~/.zshrc`. The error still appeared.

### What Worked

The problem is another file loads before `~/.zshrc`, so the fix is too late. You have to put the command in the file that loads first: `~/.zshenv`.

Run this one command in your terminal:

`echo 'ZSH_DISABLE_COMPFIX=true' >> ~/.zshenv`

Restart your terminal. The error is gone.
